itest/rust/src/register_tests/func_test.rs

```diff
@@ -148,3 +207,51 @@ impl RefCountedVirtual for GdSelfReference {
         }
     }
 }
+
+#[itest]
+fn cfg_doesnt_interfere_with_valid_method_impls() {
+    // if we re-implement this method but the re-implementation is removed, that should keep the non-removed
+    // implementation.
+    let object = Gd::new(FuncRename);
+    assert_eq!(
+        object.bind().returns_hello_world(),
+        GodotString::from("Hello world!")
+    );
+    assert_eq!(
+        object.bind().returns_bye_world(),
+        GodotString::from("Bye world!")
+    );
+}
+
+#[itest]
+fn cfg_removes_or_keeps_methods() {
+    let has_method = |name: &str| {
+        ClassDb::singleton()
+            .class_has_method_ex(GdSelfReference::class_name().to_string_name(), name.into())
+            .no_inheritance(true)
+            .done()
+    };
```

Comment:
The `has_constant`, `has_method` and `has_constant` could probably be moved out to the global scope, with the class itself being a generic parameter. It's quite likely that we may want to use them elsewhere 🙂

Reply:
Done!